src/client/providers/execInTerminalProvider.ts

```diff
@@ -38,8 +40,16 @@ function execInTerminal(fileUri?: vscode.Uri) {
         filePath = `"${filePath}"`;
     }
     const terminal = vscode.window.createTerminal(`Python`);
-    terminal.sendText(`${currentPythonPath} ${filePath}`);
-
+    if (pythonSettings.terminal && pythonSettings.terminal.executeInFileDir) {
+        const fileDirPath = path.dirname(filePath).substring(1);
+        if (fileDirPath != vscode.workspace.rootPath) {
+            terminal.sendText(`cd "${fileDirPath}"`);
+        }
+    }
+    const launchArgs = settings.PythonSettings.getInstance().terminal.launchArgs;
+    const launchArgsString = launchArgs.length > 0 ? " ".concat(launchArgs.join(" ")) : "";
+    terminal.sendText(`${currentPythonPath} ${filePath}${launchArgsString}`);
```

Comment:
Yes, thanks for spotting my crappy code. I actually did test the code, and figured that it was normal behavior for the custom arguments to print nothing to stdout. :-/